src/org/exist/Indexer.java

```diff
@@ -305,6 +305,9 @@ public void endDTD() {
     @Override
     public void endDocument() {
         if (!validate) {
+            if(indexListener != null) {
+                indexListener.endDocument(transaction);
```

Comment:
https://github.com/shabanovd/exist/blob/2.1.0-RC54/src/org/exist/indexing/AbstractStreamListener.java#L86 and https://github.com/shabanovd/exist/blob/2.1.0-RC54/src/org/exist/indexing/AbstractStreamListener.java#L93